File: 2572-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int i=0,j=0,n=s.length(),m=t.length();
        for(i=0;i<n && j<m;i++)
        {
            if(s.charAt(i)==t.charAt(j))
            {
                j++;
            }
        }
        //System.out.println(j);
        return (m-j);
    }
}

File: 2572-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.cpp
class Solution {
public:
    int appendCharacters(string s, string t) {
        int i=0,j=0,n=s.size(),m=t.size();
        for(;i<n && j<m;i++)
        {
            if(s[i]==t[j])
            j++;
        }
        return (m-j);
    }
};